src/IO/AsciiPCLoader/asciipcloader.cpp

```diff
@@ -0,0 +1,195 @@
+
+vector<string> AsciiPointCloudLoader::getFileExtensions() const {
+    return vector<string>( { "*." + asciiExt } );
+}
+
+bool AsciiPointCloudLoader::handleFileExtension( const string& extension ) const {
+    return extension.compare( asciiExt ) == 0;
+}
+
+Ra::Core::Asset::FileData* AsciiPointCloudLoader::loadFile( const string& filename ) {
+    using Ra::Core::Asset::FileData;
+    using Ra::Core::Asset::GeometryData;
+    using Ra::Core::Utils::logERROR;
+    using Ra::Core::Utils::logINFO;
+
+    FileData* fileData = new Ra::Core::Asset::FileData( filename );
+
+    if ( !fileData->isInitialized() ) {
+        delete fileData;
+        LOG( logINFO ) << "Filedata could not be initialized";
+        return nullptr;
+    }
+
+    auto startTime = clock();
+    // a unique name is required by the component messaging system
+    static int id = 0;
+    auto geometry =
+        make_unique<GeometryData>( "TRAJ_PC_" + to_string( ++id ), GeometryData::POINT_CLOUD );
+    if ( geometry == nullptr ) {
+        LOG( logERROR ) << "could not create geometry";
+        return nullptr;
+    }
+
+    geometry->setFrame( Ra::Core::Transform::Identity() );
+
+    ifstream stream( filename );
+    if ( !stream.is_open() ) {
+        throw runtime_error( "Could not open ifstream to path " + filename );
+        return nullptr;
+    }
+    else {
+        LOG( logINFO ) << "---Beginning file loading---";
+    }
+
+    /*reading how data is arranged*/
+    string line;
+
+    if ( !getline( stream, line ) ) {
+        delete fileData;
+        LOG( logINFO ) << "file does not contain trajectory data format. aborting";
+        return nullptr;
+    }
+
+    // retrieving all properties from header line
+    std::stringstream words( line );
+    vector<string> tokens;
+    string temp;
+    while ( words >> temp ) {
+        tokens.emplace_back( temp );
+    };
+
+    // making sure X, Y, Z and Time properties are found
+    int xpos = -1, ypos = -1, zpos = -1, timepos = -1;
+    for ( size_t i = 0; i < tokens.size(); ++i ) {
+        if ( xpos == -1 ) {
+            if ( tokens[i].compare( "X" ) == 0 ) { xpos = i; }
+        }
+        if ( ypos == -1 ) {
+            if ( tokens[i].compare( "Y" ) == 0 ) { ypos = i; }
+        }
+        if ( zpos == -1 ) {
+            if ( tokens[i].compare( "Z" ) == 0 ) { zpos = i; }
+        }
+        if ( timepos == -1 ) {
+            if ( tokens[i].compare( "Time" ) == 0 ) { timepos = i; }
+        }
+    }
+
+    LOG( logINFO ) << xpos << " " << ypos << " " << zpos << " " << timepos;
+
+    if ( ( xpos == -1 ) || ( ypos == -1 ) || ( zpos == -1 ) || ( timepos == -1 ) ) {
+        delete fileData;
+        LOG( logERROR ) << "file does not contain mandatory properties (X, Y, Z, Time). aborting";
+        return nullptr;
+    }
+    else {
+        LOG( logINFO ) << "loading properties: ";
+        string info = "";
+        for ( size_t i = 0; i < tokens.size(); ++i ) {
+            if ( i % 4 == 3 ) {
```

Comment:
I don't understand what is happening here

Reply:
this is just a console log, display of loaded attributes (4 in one line). It can be removed

Reply:
Yes please.

Reply:
done